### PR TITLE
Feat/update validator settings

### LINKED ISSUE #773: New storagesc update_validator_settings endpoint
We need a new storage smart contract endpoint, update_validator_settings. This will act in much the same way as [update_blobber_settings](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/blobber.go#L224) but will update a validator's stake pool settings. 

Related to https://github.com/0chain/0chain/issues/772

### PR DESCRIPTION
**Pull request by lawRathod**
## Changes
* Adds new method `getValidator` on `StorageSmartContract`
* Adds new method `emitUpdateValidator` in validator_eventdb.go 
* New tag for events process `TagValidatorUpdated`
* Adds support for `updateValidatorSettings`
* Adds `updateValidator` on `StorageSmartContract`
* Adds benchmark test for `updateValidatorSettings`
* Adds `update_validator_settings` in sc.yaml to `storagesc.cost`

Closes #773

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/validator_eventdb.go [hunk 1]
```diff
@@ -54,6 +55,25 @@ func emitAddOrOverwriteValidatorTable(vn *ValidationNode, balances cstate.StateC
...
+			"min_stake":       int64(sn.StakePoolSettings.MinStake),
+			"max_stake":       int64(sn.StakePoolSettings.MaxStake),
+			"num_delegates":   sn.StakePoolSettings.MaxNumDelegates,
+			"service_charge":  sn.StakePoolSettings.ServiceChargeRatio,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("marshalling update: %v", err)
+	}
+	balances.EmitEvent(event.TypeStats, event.TagUpdateBlobber, sn.ID, string(data))
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/validator_eventdb.go]: Event should be `TagUpdateValidator`

**Review by peterlimg — CHANGES_REQUESTED** [review]
Please check [updateBlobberSettings](https://github.com/0chain/0chain/blob/4f4f4af566256307530e45428c311810f4f0efdc/code/go/0chain.net/smartcontract/storagesc/blobber.go#L80) for how to update baseURL, we need to make sure the URL is unique.

**Comment by Sriep** [comment]
We need changes to GoSDK and zbox, to access this new endpoint. This works the same way as update-blobber.

File: code/go/0chain.net/smartcontract/storagesc/validator_eventdb.go (hunk 1 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/validator_eventdb.go]: Fixed!

File: code/go/0chain.net/smartcontract/dbs/event/process.go [hunk 2]
```diff
@@ -36,6 +36,8 @@ const (
 	TagAddWriteMarker
 	TagAddBlock
 	TagAddOrOverwriteValidator
+	TagUpdateValidator
+	TagAddOrOverwriteReadMarker
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/dbs/event/process.go]: Unnecessary tag `TagAddOrOverwriteReadMarker` for this PR.

File: code/go/0chain.net/smartcontract/storagesc/config_settigns.go [hunk 3]
```diff
@@ -295,6 +297,7 @@ var (
 		"cost.challenge_response":          {CostChallengeResponse, smartcontract.Cost},
 		"cost.generate_challenges":         {CostGenerateChallenges, smartcontract.Cost},
 		"cost.add_validator":               {CostAddValidator, smartcontract.Cost},
+		"cost.update_validator_settings":   {CostUpdateValidatorSettings, smartcontract.Cost},
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/config_settigns.go:300]: Update sc.yaml to add the costs in `storagesc.cost` field

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/config_settigns.go (hunk 3 above)
Comment (dabasov) [reply code/go/0chain.net/smartcontract/storagesc/config_settigns.go:300]: use 100 as default

File: code/go/0chain.net/smartcontract/dbs/event/process.go (hunk 2 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/dbs/event/process.go]: I wonder how did it get there. Fixed in future commit.

File: code/go/0chain.net/smartcontract/storagesc/config_settigns.go (hunk 3 above)
Comment (lawRathod) [reply code/go/0chain.net/smartcontract/storagesc/config_settigns.go:300]: Done

Comment (dabasov) [reply code/go/0chain.net/smartcontract/storagesc/config_settigns.go:300]: @lawRathod to run system-tests with updated sc.yaml, you need to hack actions a little, please reach Kishan for help

**Review by dabasov — APPROVED** [review]
lgtm

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**